sdk/python/foundation-models/system/finetune/chat-completion/chat-completion.ipynb

```diff
@@ -0,0 +1,685 @@
+    "    )\n",
+    "\n",
+    "if computes_allow_list is not None:\n",
+    "    computes_allow_list_lower_case = [x.lower() for x in computes_allow_list]\n",
+    "    if compute.size.lower() not in computes_allow_list_lower_case:\n",
+    "        raise ValueError(\n",
+    "            f\"VM size {compute.size} is not in the allow-listed computes for finetuning\"\n",
+    "        )\n",
+    "else:\n",
+    "    # Computes with K80 GPUs are not supported\n",
+    "    unsupported_gpu_vm_list = [\n",
+    "        \"standard_nc6\",\n",
+    "        \"standard_nc12\",\n",
+    "        \"standard_nc24\",\n",
+    "        \"standard_nc24r\",\n",
+    "    ]\n",
+    "    if compute.size.lower() in unsupported_gpu_vm_list:\n",
+    "        raise ValueError(\n",
+    "            f\"VM size {compute.size} is currently not supported for finetuning\"\n",
+    "        )\n",
+    "\n",
+    "\n",
+    "# This is the number of GPUs in a single node of the selected 'vm_size' compute.\n",
+    "# Setting this to less than the number of GPUs will result in underutilized GPUs, taking longer to train.\n",
+    "# Setting this to more than the number of GPUs will result in an error.\n",
+    "gpu_count_found = False\n",
+    "workspace_compute_sku_list = workspace_ml_client.compute.list_sizes()\n",
+    "available_sku_sizes = []\n",
+    "for compute_sku in workspace_compute_sku_list:\n",
+    "    available_sku_sizes.append(compute_sku.name)\n",
+    "    if compute_sku.name.lower() == compute.size.lower():\n",
+    "        gpus_per_node = compute_sku.gpus\n",
+    "        gpu_count_found = True\n",
+    "# if gpu_count_found not found, then print an error\n",
+    "if gpu_count_found:\n",
+    "    print(f\"Number of GPU's in compute {compute.size}: {gpus_per_node}\")\n",
+    "else:\n",
+    "    raise ValueError(\n",
+    "        f\"Number of GPU's in compute {compute.size} not found. Available skus are: {available_sku_sizes}.\"\n",
+    "        f\"This should not happen. Please check the selected compute cluster: {compute_cluster} and try again.\"\n",
+    "    )"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4. Pick the dataset for fine-tuning the model\n",
+    "\n",
+    "We use the [ultrachat_200k](https://huggingface.co/datasets/samsum) dataset. The next few cells show basic data preparation for fine tuning:\n",
+    "* Visualize some data rows\n",
+    "* Preprocess the data and format it in required format."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# download the dataset using the helper script. This needs datasets library: https://pypi.org/project/datasets/\n",
+    "import os\n",
+    "\n",
+    "exit_status = os.system(\n",
+    "    \"python ./download-dataset.py --dataset HuggingFaceH4/ultrachat_200k --download_dir ultrachat_200k_dataset --dataset_split_pc 5\"\n",
```

Comment:
Maybe we can add a description saying that "Running the below script only downloads 5% of the data. you can increase the `dataset_split_pc` to change this parameter"